### PR TITLE
[4.1] Add server timing response header in debug mode

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Browsers do support a [server timing](https://www.w3.org/TR/server-timing/#the-server-timing-header-field.) header field which adds information what happens on the server side. This basically gives the developer insight what happens in the "Time to First Byte".
This pr adds the missing header in the debug plugin, as this header should not be added when in production mode because of performance reasons.

Would be nice to get some feedback about spicing up the values of the header as currently I'm just logging them as they are.

The view looks like in the following picture:
image

### Testing Instructions
- When testing in Firefox, make sure you open the page with HTTPS
- Put joomla in debug mode
- In the debug plugin with make sure that the Profile collector is enabled (is by default)
- Open any page
- Open the dev tools and check the Network tab
- Select the first entry and then click on the timings tab of the specific request

### Actual result BEFORE applying this Pull Request
Only the request timing information is shown.

### Expected result AFTER applying this Pull Request
Beside the request timing is the server timing information shown.

### CODE REVIEW

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on cf42d70171e39e69d25c9e67b2a1fe56911520c3<br /><br />@bembelimen this would be great Dev DX if could land in 4.1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36231">issues.joomla.org/tracker/joomla-cms/36231</a>.</sub>

**Comment by Quy** [comment]
Maybe add a colon after each description for consistency.

**Comment by laoneo** [comment]
@Quy that's how it comes from the profiler. How would you do it then, can you make a suggestion?

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -665,4 +666,38 @@ private function collectLogs(): self
...
+			if (stripos($mark->label, 'before') !== false)
+			{
+				continue;
+			}
+
+			$desc = $mark->label;
+			$desc = str_ireplace('after', '', $desc);
+
+			$name     = preg_replace('/[^\da-z]/i', '', $desc);
+			$metrics .= sprintf('%s;dur=%f;desc="%s", ', $index . $name, $mark->time, $desc);
```
Comment (Quy) [inline plugins/system/debug/debug.php]:
```suggestion
			$metrics .= sprintf('%s;dur=%f;desc="%s:", ', $index . $name, $mark->time, $desc);
```
Maybe?

Comment (laoneo) [reply plugins/system/debug/debug.php]: Now I got you

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 27300cf8bfc0c3d626634dd9809dc431b2e6a8f6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36231">issues.joomla.org/tracker/joomla-cms/36231</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36231">issues.joomla.org/tracker/joomla-cms/36231</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**